packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -44,7 +45,20 @@ const LinkControlSearchInput = ( {
 	}
 
 	return (
-		<form onSubmit={ selectSuggestionOrCurrentInputValue }>
+		// eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+		<form
+			onSubmit={ selectSuggestionOrCurrentInputValue }
+			onKeyDown={ ( event ) => {
+				event.stopPropagation();
```

Comment:
I'm trying to work out why this is necessary. As this reintroduces code that was removed could we consider a comment here to understand why it's necessary?

Reply:
See previous conversation.